### PR TITLE
:sparkles: initial commit for teensy 4.0 and 4.1 boards

### PR DESCRIPTION
**Pull request by acarrou**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -9,16 +9,16 @@
 required_conan_version = ">=1.50.0"
 
 
-class LibhalLibraryConan(ConanFile):
-    name = "libhal-library"
+class LibhalTeensyConan(ConanFile):
```
Comment (kammce) [inline conanfile.py:12]: Do not call this Teensy. The chip exists outside of the teensy project line. Stick with IMX. Do the same for the whole file except for things that are specifically for teensy. Like you can mention it in the tags.

File: conanfile.py [hunk 2]
```diff
@@ -78,13 +79,71 @@ def package(self):
...
+
+            # List of REQUIRED compiler flags for the gnu-arm-embedded-toolchain for some
+            # of the chips. These are determined by the capabilities of the chip.
+            # For example all but the teensy4072 and teensy4074 have hardware floating point
+            # arithmetic support so they ought to use "float-abi=softfp" which uses the
+            # floating point hardware BUT is ABI compatible with the software
+            # implementation.
+            m7_architecture_flags = [
+                "-mcpu=cortex-m7",
+                "-mfloat-abi=hard",
```
Comment (kammce) [inline conanfile.py:101]: Never use hard, use softfp for hard floating point with with software compatibility.

File: conanfile.py [hunk 3]
```diff
@@ -78,13 +79,71 @@ def package(self):
...
+            # package's package_folder.
+            linker_path = os.path.join(self.package_folder, "linker_scripts")
+
+            # Set the cmake file name
+            self.cpp_info.set_property("cmake_file_name", "libhal-i.mx")
+            # All the package to be found in anyway with cmake
+            self.cpp_info.set_property("cmake_find_mode", "both")
+
+            # Create the special/generic component "teensy" and set its component name
+            self.cpp_info.components["teensy"].set_property(
```
Comment (kammce) [inline conanfile.py:114]: Make the component imx

File: conanfile.py [hunk 4]
```diff
@@ -78,13 +79,71 @@ def package(self):
...
+                # compiler flags
+                self.cpp_info.components[component].exelinkflags.append(link_script)
+                self.cpp_info.components[component].exelinkflags.extend(flags)
+                # Add flags to the cflags & cxxflags to ensure that each compilation unit
+                # Knows the instruction set and float ABI
+                self.cpp_info.components[component].cflags = flags
+                self.cpp_info.components[component].cxxflags = flags
+
+            # Create the components for each chip.
+            create_component(self, "imx-rt1062-teensy4", m7_architecture_flags)
```
Comment (kammce) [inline conanfile.py:144]: Now that I think about it, we can shorten this to just teensy4 and teensy41. We don't need to further qualify it with redundant information.

**Review by kammce — CHANGES_REQUESTED** [review]

File: conanfile.py [hunk 5]
```diff
@@ -9,14 +9,14 @@
...
+class LibhalIMXConan(ConanFile):
     name = "libhal-i.mx"
     version = "0.1.0"
     license = "Apache-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/libhal/libhal-i.mx"
-    description = ("A collection of drivers for the teensy")
-    topics = ("teensy", "libhal", "driver")
+    description = ("A collection of drivers for the imx")
+    topics = ("imx", "libhal", "driver")
```
Comment (kammce) [inline conanfile.py:19]: Add back teensy. I said that's still relevant here because the imx is used by them. Just shouldn't be the name for everything.

**Review by kammce — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 5 above)
Comment (acarrou) [reply conanfile.py]: sorry about that, CTRL+D too much 😄

File: conanfile.py [hunk 6]
```diff
@@ -15,8 +15,8 @@ class LibhalIMXConan(ConanFile):
     license = "Apache-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/libhal/libhal-i.mx"
-    description = ("A collection of drivers for the imx")
-    topics = ("imx", "libhal", "driver")
+    description = ("A collection of drivers for the teensy")
+    topics = ("teensy", "libhal", "driver")
```
Comment (kammce) [inline conanfile.py:19]: Add both lol

**Review by kammce — CHANGES_REQUESTED** [review]

**Review by kammce — APPROVED** [review]
